### PR TITLE
Bedrock ledger upgrade

### PR DESCRIPTION
**Pull request by aljones15**
In this PR
- [x] this takes the zcap action changes PR and just in general improves test coverage, deps, etc.
- [x] adds `invocationTarget` to capability proof.
- [x] changes capabilityAction to `write`
- [x] reintroduces a minimal working test project
- [x] removes mongodb usernames and passwords
- [x] adds github actions with automated tests
- [x] upgrades to `zcapld: 5.0.0`
- [x] removes bedrock-identity and replaces with `bedrock-account`
- [x] fix 1 broken test.

Tests:
1. All tests in the `/test` dir pass
2. project starts up with no errors
3. ledger in db is there with a valid ledger-agent

NOTE: this contains a number of improvements to this PR:
https://github.com/veres-one/veres-one/pull/49

### CODE REVIEW

File: lib/plugins/ticketService.js
Comment (dlongley) [inline lib/plugins/ticketService.js]: Please fix this too.

File: lib/ledger.js
Comment (dlongley) [inline lib/ledger.js]: A ledger would never have an address like this -- this is a zcap identifier. This identifier (well, `urn:zcap:root:encodeURIComponent(ledger ID)`) would be used as the ledger's root capability ID, it would not ever be an invocation target. The ledger ID itself would be used in the invocation target field.

We shouldn't put this here -- even as a mock value -- because it will create more confusion for whomever works on changing this mock proof to a real one. Can you find and put the ledger ID (you should be able to find it in the ledger config, wherever that is) here instead? If the ledger ID is auto-generated later or something, then just auto-generate a mock one and put it here.

File: test/mocha/40-config-change.js
Comment (dlongley) [inline test/mocha/40-config-change.js]:
```suggestion
// a config queue (like an operation queue) might make it possible in the future
```

File: configs/dev-genesis.js
Comment (dlongley) [inline configs/dev-genesis.js:48]: I don't know why these things are commented out. Can you leave a comment or fix it?

File: configs/production.js.save
Comment (dlongley) [inline configs/production.js.save:54]: I don't know what's going on here, can you leave a comment or fix it?

File: configs/testnet.js
Comment (dlongley) [inline configs/testnet.js:58]: Same issue here -- needs comments/fixes.

File: lib/ledger.js
Comment (dlongley) [inline lib/ledger.js]: This needs at least a FIXME. We shouldn't generate a seed from a short maintainer passphrase. We either need a very long passphrase that we hash appropriately or we need to use a salt+b/scrypt -- if we're going to produce a seed from human readable text.

**Review by dlongley — CHANGES_REQUESTED** [review]

File: configs/dev-genesis.js
Comment (aljones15) [reply configs/dev-genesis.js:48]: this is old actually. When we updated to the latest version of mongodb it turns out that trying to login to a mongodb server with out auth enabled is fatal. Hence only add username, password, and authSource if you have a valid username, password, etc. already setup.

comments added here: https://github.com/veres-one/veres-one/pull/44/commits/551293071163a115939c7319c3ad5f22b69292d0

File: configs/testnet.js
Comment (aljones15) [reply configs/testnet.js:58]: Added comments on mongodb auth options here: https://github.com/veres-one/veres-one/pull/44/commits/551293071163a115939c7319c3ad5f22b69292d0

File: configs/production.js.save
Comment (aljones15) [reply configs/production.js.save:54]: added comments on mongodb auth options here: https://github.com/veres-one/veres-one/pull/44/commits/551293071163a115939c7319c3ad5f22b69292d0

File: lib/ledger.js
Comment (aljones15) [reply lib/ledger.js]: changed it to use a `sha-256` hash: https://github.com/veres-one/veres-one/pull/44/commits/01b3c12a00d416d83f10d62240bf54000a07722a

File: lib/ledger.js
Comment (aljones15) [reply lib/ledger.js]: let me know if this method is correct: https://github.com/veres-one/veres-one/pull/44/commits/4d17de37351a3f067d74dc016337b420f9563659

File: lib/plugins/ticketService.js
Comment (aljones15) [reply lib/plugins/ticketService.js]: replaced it with a mock did here: https://github.com/veres-one/veres-one/pull/44/commits/4d17de37351a3f067d74dc016337b420f9563659

File: configs/dev-genesis.js
Comment (dlongley) [inline configs/dev-genesis.js:48]: The default for code that isn't used is to remove it. If it sticks around in commented form, we need a comment that explains both why it's commented and why it wasn't just removed. The above comment just makes me wonder:

1. Why don't we want it to try to login? Is that temporary?
2. Is it supposed to try and login sometimes and I may want these lines here to more easily accomplish that?
3. How is this situation different for veres-one vs. other bedrock apps -- such that these lines should be here as commented? (Why the inconsistency, what's special?)

So, can you add the reason why these were not removed? Same here as for any other commented code.

File: configs/secrets/dev-genesis-secrets.js
Comment (dlongley) [inline configs/secrets/dev-genesis-secrets.js]:
```suggestion
// FIXME remove this password entirely and replace it with a temporary key
// pair that MUST be replaced in production but is only used in dev mode
// consider using `ensureOverride` feature,
// see: https://github.com/digitalbazaar/bedrock/blob/main/lib/config.js#L61
```

File: lib/ledger.js
Comment (dlongley) [inline lib/ledger.js:153]:
```suggestion
    // FIXME: do not use passwords at all, use a key pair
    // that must be provided/overridden in production mode

    // creates 32 byte hashes for passwords
```

File: lib/ledger.js
Comment (dlongley) [inline lib/ledger.js]: This is unnecessarily complex. The function that calls `_sendElectorPool` can just pass in the ledger ID. I see it's called from the function that sets it here:

https://github.com/veres-one/veres-one/blob/4d17de37351a3f067d74dc016337b420f9563659/lib/ledger.js#L188

So just add a `ledgerId` param to `_sendElectorPool` and pass it in.

```suggestion
```

**Review by dlongley — CHANGES_REQUESTED** [review]

File: configs/dev-genesis.js
Comment (aljones15) [reply configs/dev-genesis.js:48]: I would leave it there in case someone wants to try a development build with auth enabled.

Comment (dlongley) [reply configs/dev-genesis.js:48]: Great! If you add that as a comment to the code I think it would answer my questions (and others when they read it too).

File: configs/dev-genesis.js
Comment (dlongley) [inline configs/dev-genesis.js]:
```suggestion
// they are left commented here for convenience; you can uncomment them
// if you want to try a dev build with auth enabled
```

File: lib/ledger.js
Comment (dlongley) [inline lib/ledger.js]: Don't do this. Pass in the ledger ID. If you do this, when someone fixes/changes the ledger ID to be something else, they will miss this and we'll have a bug on our hands.

Comment (aljones15) [reply lib/ledger.js]: good idea and done here: https://github.com/veres-one/veres-one/pull/44/commits/113e33310996b7596530ea90364b74a1abacce90

File: configs/dev-genesis.js
Comment (aljones15) [reply configs/dev-genesis.js:48]: comments were added addressing this.

**Review by dlongley — APPROVED** [review]
Thanks! There's still more work to do here once this is merged, but this is a good improvement.

**Comment by msporny** [comment]
I've fixed all of this up and confirmed that you can run the ledger, add a DID and that the tests work.

**Pull request merged**
